tracing/tracer_server.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 CloudWeGo Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/pkg/stats"
+	"github.com/cloudwego/kitex/server"
+	"github.com/kitex-contrib/obs-opentelemetry/tracing/internal"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	oteltrace "go.opentelemetry.io/otel/trace"
+)
+
+var _ stats.Tracer = (*serverTracer)(nil)
+
+type serverTracer struct {
+	config            *config
+	histogramRecorder map[string]metric.Float64Histogram
+}
+
+func newServerOption(opts ...Option) (server.Option, *config) {
+	cfg := newConfig(opts)
+	st := &serverTracer{
+		config: cfg,
+	}
+
+	st.createMeasures()
+
+	return server.WithTracer(st), cfg
+}
+
+func (s *serverTracer) createMeasures() {
+	serverDurationMeasure, err := s.config.meter.NewFloat64Histogram(ServerDuration)
+	handleErr(err)
+
+	s.histogramRecorder = map[string]metric.Float64Histogram{
+		ServerDuration: serverDurationMeasure,
+	}
+}
+
+func (s *serverTracer) Start(ctx context.Context) context.Context {
+	tc := &internal.TraceCarrier{}
```

Comment:
> s.config.spanNameFormatter should be set to TraceCarrier here

Since we can get the spanNameFormatter directly from the config, it feels like we don't need to put the spanNameFormatter in the carrier?